boot-cw-plus/src/contracts/cw3_flex_multisig.rs

```diff
@@ -7,7 +7,7 @@ pub use cw3_flex_multisig::msg::{
     ExecuteMsgFns as Cw3FlexMultisigExecuteMsgFns, QueryMsgFns as Cw3FlexMultisigQueryMsgFns,
 };
 
-#[contract(InstantiateMsg, ExecuteMsg, QueryMsg, Empty)]
```

Comment:
I'm wondering, is there a way to keep the ExecuteMsg generic here? That would be impossible with our current traits right?

Reply:
I don't really see how it could be compatible with  the current traits. We assume, as every contract, that the types used for the messages don't have generics. I don't see the use case for a generic parameter here ?

Reply:
Yeah you're right

Reply:
Ok, so this seems possible to create such a contract structure for later instantiation : Something like this for a single generic parameter and could be adapted for more type parameters

```rust
pub struct GenericTestContract<Chain: ::boot_core::CwEnv, T>(::boot_core::Contract<Chain>, Option<std::marker::PhantomData<T>>);
```

Reply:
Hmm yeah, and the CwInterface 

```rust
impl <Chain, T> CwInterface for GenericTestContract<chain, T> {
    type ExecuteMsg = ExecuteMsg<T>
}
```
Does that work?

Reply:
Even if it would work I think we should keep it for another PR. 👍

Reply:
Yes this works : 
![image](https://user-images.githubusercontent.com/44806566/234249178-b6071a68-e791-46b5-8a1e-2b6b6b74d3c4.png)

Reply:
can you try adding a default to T?
`ExecuteMsg<T = Empty>`
That way we don't always have to provide the generic

Reply:
Sure, this was only a macro expansion to show you what code it would need to generate to make that work